src/Employer/Employer.Web/Controllers/ErrorController.cs

```diff
@@ -24,29 +24,52 @@ public ErrorController(ILogger<ErrorController> logger)
         [Route("error/{id?}")]
         public IActionResult Error(int id)
         {
+            switch (id)
+            {
+                case 403:
+                    return AccessDenied();
+                default:
+                    break;
+            }
+
             return View(new ErrorViewModel { StatusCode = id, RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
         }
 
         [Route("error/handle")]
         public IActionResult ErrorHandler()
         {
+            if (HttpContext.Items.TryGetValue(ContextItemKeys.EmployerIdentifier, out var accountId))
+            {
+                ViewBag.EmployerAccountId = accountId;
+            }
+
             var exceptionFeature = HttpContext.Features.Get<IExceptionHandlerPathFeature>();
 
             if (exceptionFeature != null)
             {
-                var accountId = (string)HttpContext.Items[ContextItemKeys.EmployerIdentifier];
                 string routeWhereExceptionOccurred = exceptionFeature.Path;
                 var exception = exceptionFeature.Error;
 
                 if (exception is InvalidStateException)
                 {
-                    _logger.LogError(exception, $"Exception on path: {routeWhereExceptionOccurred}");                    
+                    _logger.LogError(exception, $"Exception on path: {routeWhereExceptionOccurred}");
                     TempData.Add(TempDataKeys.DashboardErrorMessage, exception.Message);
                     return RedirectToRoute(RouteNames.Dashboard_Index_Get, new { EmployerAccountId = accountId });
                 }
+
+                if (exception is AuthorisationException)
+                {
+                    _logger.LogError(exception, exception.Message);

```

Comment:
Should this be logged as a warning as it's not really an application failure and it's a handled scenario?

Reply:
👍